_posts/2025-03-27-Boost-OpenSearch-VectorSearch-Performance-With-Intel-AVX512.md

```diff
@@ -0,0 +1,193 @@
+
+### **Inner Product Space Type**
+
+* **FP32 Encoding:**  
+  * Hot functions:  
+    * *faiss::fvec\_inner\_product*   
+    * *faiss::fvec\_inner\_product\_batch\_4*.  
+  * Benefits of AVX-512:  
+    * Indexing: Up to 75% reduction in cycles.  
+    * Search: Up to 8% reduction in cycles.  
+* **SQFP16 Encoding:**  
+  * Hot function: *faiss::query\_to\_code*.  
+  * Benefits of AVX-512:  
+    * Indexing: Up to 39% reduction in cycles.  
+    * Search: Up to 11% reduction in cycles.
+
+### **L2 (euclidean) Space Type**
+
+* **FP32 Encoding:**  
+  * Hot functions:   
+    * *faiss::fvec\_L2sqr*   
+    * *faiss::fvec\_L2sqrt\_batch\_4*.  
+  * Benefits of AVX-512:  
+    * Indexing: Up to 54% reduction in cycles.  
+    * Search: Up to 11% reduction in cycles.  
+* **SQFP16 Encoding:**  
+  * Hot function: *faiss::query\_to\_code*.  
+  * Benefits of AVX-512:  
+    * Indexing: Up to 17% reduction in cycles.  
+    * Search: Up to 6% reduction in cycles.
+
+| Inner Product space type |  |  |  |  |
+| ----- | :---- | :---- | :---- | :---- |
+|  | Encoding | Function | %Cycle Spent (AVX2) | %Cycle Spent (AVX512) |
+| Indexing | fp32 | fvec\_inner\_product | 28.86 | 7.32 |
+| Indexing | SQfp16 | query\_to\_code | 17.95 | 10.94 |
+| Search | fp32 | fvec\_inner\_product\_batch\_4 | 34.66 | 31.74 |
+| Search | SQfp16 | query\_to\_code | 42.24 | 37.73 |
+
+| L2 space type |  |  |  |  |
+| ----- | :---- | :---- | :---- | :---- |
+|  | Encoding | Function | %Cycle Spent (AVX2) | %Cycle Spent (AVX512) |
+| Indexing | fp32 | fvec\_L2sqr | 36.76 | 16.75 |
+| Indexing | SQfp16 | query\_to\_code | 26.18 | 21.61 |
+| Search | fp32 | fvec\_L2sqr\_bactch\_4 | 31.80 | 28.32 |
+| Search | SQfp16 | query\_to\_code | 36.99 | 34.72 |
+
+Starting with OpenSearch version 2.18, AVX-512 is enabled by default. As of March 2025, OpenSearch has shown best performance for AVX-512 on AWS r7i instances. 
+
+The next section describes benchmark results that were run with AVX2 and AVX-512 versions of the Faiss library shipped for x64 architecture ([Documentation](https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#supported-lucene-methods:~:text=multiple%20of%208.-,x64%20architecture,libopensearchknn_faiss.so%3A%20The%20non%2Doptimized%20Faiss%20library%20without%20SIMD%20instructions.,-When%20using%20the)). These benchmarks were run using [OpenSearch Benchmark (OSB) vector search workload](https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/main/vectorsearch) using configuration in appendix1.
```

Comment:
@assanedi we don't need appendix section anymore, directly link to that github issue comment here where you added the configuration